src/ops.rs

```diff
@@ -127,6 +127,142 @@ pub unsafe fn add_n_m_itch(n: usize, m: usize) -> usize {
+/// * Size of `output` must be exactly `n+m` limbs. `output` may be filled with arbitrary data.
+/// * Size of `scratch_space` must be exactly [`mul_n_m_itch(n, m)`](mul_n_m_itch) limbs.
+///   `scratch_space` may be filled with arbitrary data.
+///
+/// ## Guarantees
+/// Function is guaranteed to be constant time in size of its arguments. Ie. the amount of instructions
+/// and memory access patterns should be the same for all numbers with the same `n` and `m`.
+///
+/// ## Algorithm
+/// This method calls the [mpn_sec_mul](gmp::mpn_sec_mul) function.
+pub unsafe fn mul_n_m(a: &[Digit], b: &[Digit], scratch_space: &mut [Digit], output: &mut [Digit]) {
+    debug_assert!(a.len() >= b.len());
+    debug_assert!(!b.is_empty());
+    debug_assert_eq!(output.len(), a.len() + b.len());
+    debug_assert_eq!(scratch_space.len(), mul_n_m_itch(a.len(), b.len()));
+
+    let n = a.len() as size_t;
+    let m = b.len() as size_t;
+    let pa = a.as_ptr() as *const limb_t;
+    let pb = b.as_ptr() as *const limb_t;
+    let pout = output.as_mut_ptr() as *mut limb_t;
+    let scratch_space_new = scratch_space.as_mut_ptr() as *mut limb_t;
+
+    gmp::mpn_sec_mul(pout, pa, n, pb, m, scratch_space_new);
+}
+
+/// Returns size of scratch space required for [`mpn_sec_mul`](gmp::mpn_sec_mul) function.
+///
+/// ## Safety
+/// Requires that `n` is greater or equal than `m`, ie `n >= m`
+pub unsafe fn mul_n_m_itch(n: usize, m: usize) -> usize {
+    gmp::mpn_sec_mul_itch(n as _, m as _) as _
+}
+
+/// Takes two integers `a` and `b` represented with sequence of limbs of size `n` and `m` in order
+/// from least to most significant. Computes floor of the division `floor(a/b)` and writes it to `output`,
+/// also writing the remainder `a mod b` to the most significant `m` limbs of `a`. Accordingly, `a` must be mutable.
+///
+/// ## Safety
+/// Following requirements must be met:
+/// * `n >= m > 0`
+///   Ie. size of `a` must be greater than `b`. It does not imply that `a > b` since both numbers
+///   can be padded with zeroes. `b` must not be empty.
+/// * The most significant limb of `b` must be non-zero.
+/// * Size of `output` must be exactly `n-m` limbs. `output` may be filled with arbitrary data.
+/// * Size of `scratch_space` must be exactly [`div_n_m_itch(n, m)`](div_n_m_itch) limbs.
+///   `scratch_space` may be filled with arbitrary data.
+///
+/// ## Guarantees
+/// Function is guaranteed to be constant time in size of its arguments. Ie. the amount of instructions
+/// and memory access patterns should be the same for all numbers with the same `n` and `m`.
+///
+/// ## Algorithm
+/// This method calls the [mpn_sec_div_qr](gmp::mpn_sec_div_qr) function.
+pub unsafe fn div_n_m(
+    a: &mut [Digit],
+    b: &[Digit],
+    scratch_space: &mut [Digit],
+    output: &mut [Digit],
+) -> Digit {
+    debug_assert!(a.len() >= b.len());
+    debug_assert!(!b.is_empty());
+    debug_assert!(b.last().unwrap().ne(&Digit::zero()));
```

Comment:
Why don't you use the standard operator?

```suggestion
    debug_assert!(b.last().unwrap() != &Digit::zero());
```

Reply:
Or, in this case, there's even better way to write assert:

```suggestion
    debug_assert_ne!(b.last(), Some(&Digit::zero()));
```